### PR TITLE
noble: Advertisement serviceData is an array

### PR DESCRIPTION
**Pull request by claytonkucera**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/noble/noble/blob/c4cd18a7a429bb832f6c4ca793be3faf9af884e9/lib/hci-socket/gap.js#L197
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by thegecko — APPROVED** [review]
LGTM

File: types/noble/index.d.ts [hunk 1]
```diff
@@ -65,10 +66,10 @@ export declare class Peripheral extends events.EventEmitter {
 
 export interface Advertisement {
     localName: string;
-    serviceData: {
+    serviceData: [{
         uuid: string,
         data: Buffer
-    };
```
Comment (bioball) [inline types/noble/index.d.ts]: Wrapping brackets around a type tells TypeScript that this is a tuple type; both its length and members are typesafe. Take a look at the playground as an example: https://www.typescriptlang.org/play/#src=declare%20class%20Buffer%20%7B%0D%0A%20%20%20%20static%20from(src%3A%20string)%3A%20Buffer%3B%0D%0A%7D%0D%0A%0D%0Atype%20ServiceData%20%3D%20%5B%7B%0D%0A%20%20uuid%3A%20string%2C%0D%0A%20%20data%3A%20Buffer%0D%0A%7D%5D%0D%0A%0D%0Aconst%20data%3A%20ServiceData%20%3D%20%5B%0D%0A%20%20%7B%20uuid%3A%20'foo'%2C%20data%3A%20Buffer.from('foo')%20%7D%2C%0D%0A%20%20%7B%20uuid%3A%20'bar'%2C%20data%3A%20Buffer.from('bar')%20%7D%0D%0A%5D

This probably should be annotated as `Array<{ uuid: string, data: Buffer }>` instead.

**Review by bioball — CHANGES_REQUESTED** [review]
Comments inline

File: types/noble/index.d.ts (hunk 1 above)
Comment (claytonkucera) [reply types/noble/index.d.ts]: Right you are @bioball ! Thanks. See my most recent commit.

**Review by bioball — APPROVED** [review]
LGTM!

**Pull request merged**